### PR TITLE
Cron: remove nextcloud-aio-domaincheck if stopped

### PR DESCRIPTION
**Pull request by Zoey2936**

### CODE REVIEW

File: Containers/mastercontainer/cron.sh [hunk 1]
```diff
@@ -47,8 +47,14 @@ while true; do
     # Remove sessions older than 24h
     find "/mnt/docker-aio-config/session/" -mindepth 1 -mmin +1440 -delete
 
+    # Remove nextcloud-aio-domaincheck container
+    sudo -u www-data docker container remove nextcloud-aio-domaincheck --force
```
Comment (szaimen) [inline Containers/mastercontainer/cron.sh:51]: we should only remove the container if it is stopped. You should be able to get it with e.g.  docker ps --filter "status=exited"

File: Containers/mastercontainer/cron.sh [hunk 2]
```diff
@@ -47,8 +47,14 @@ while true; do
...
+
     # Remove dangling images
-    sudo -u www-data docker image prune -f
+    sudo -u www-data docker image prune --force
+    
+    # Remove dangling volumes
+    for volume in $(sudo -u www-data docker volume ls --quiet --filter dangling=true | grep -E "^[a-z0-9]{64}$"); do sudo -u www-data docker volume remove "$volume"; done
```
Comment (szaimen) [inline Containers/mastercontainer/cron.sh:57]: I am honestly against removing any volumes automatically.

The better way is to remove anonymous volumes when the container gets removed which we already do in https://github.com/nextcloud/all-in-one/blob/3f22ec8b4d26cb3b8275cd7ae8a71ba2c957f067/php/src/Docker/DockerActionManager.php#L148

File: Containers/mastercontainer/cron.sh (hunk 1 above)
Comment (Zoey2936) [reply Containers/mastercontainer/cron.sh:51]:
```suggestion
    if [ "$(docker ps --format "{{.Names}}" --filter "status=exited" | grep -w "nextcloud-aio-domaincheck" | head -n 1)" == "nextcloud-aio-domaincheck" ]; then
        sudo -u www-data docker container remove nextcloud-aio-domaincheck
    fi
```

Comment (szaimen) [reply Containers/mastercontainer/cron.sh:51]:
```suggestion
    if [ "$(docker ps --format "{{.Names}}" --filter "status=exited" | grep -q "^nextcloud-aio-domaincheck$")" == "nextcloud-aio-domaincheck" ]; then
        sudo -u www-data docker container remove nextcloud-aio-domaincheck
    fi
```

Comment (Zoey2936) [reply Containers/mastercontainer/cron.sh:51]:
```suggestion
    if docker ps --format "{{.Names}}" --filter "status=exited" | grep -qw "^ttt$"; then
        sudo -u www-data docker container remove nextcloud-aio-domaincheck
    fi
```

File: Containers/mastercontainer/cron.sh (hunk 2 above)
Comment (Zoey2936) [inline Containers/mastercontainer/cron.sh:57]:
```suggestion
```

File: Containers/mastercontainer/cron.sh [hunk 3]
```diff
@@ -47,8 +47,13 @@ while true; do
     # Remove sessions older than 24h
     find "/mnt/docker-aio-config/session/" -mindepth 1 -mmin +1440 -delete
 
+    # Remove nextcloud-aio-domaincheck container
+    if docker ps --format "{{.Names}}" --filter "status=exited" | grep -qw "^nextcloud-aio-domaincheck$"; then
+        sudo -u www-data docker container remove nextcloud-aio-domaincheck
```
[the remark this answers is not in the document]
Comment (Zoey2936) [reply Containers/mastercontainer/cron.sh:52]: why not `container remove`?

[the remark this answers is not in the document]
Comment (szaimen) [reply Containers/mastercontainer/cron.sh]: was just a suggestion. container remove is also fine for me

File: Containers/mastercontainer/cron.sh [hunk 4]
```diff
@@ -47,8 +47,13 @@ while true; do
     # Remove sessions older than 24h
     find "/mnt/docker-aio-config/session/" -mindepth 1 -mmin +1440 -delete
 
+    # Remove nextcloud-aio-domaincheck container
+    if sudo -u www-data docker ps --format "{{.Names}}" --filter "status=exited" | grep -qw "^nextcloud-aio-domaincheck$"; then
```
Comment (szaimen) [inline Containers/mastercontainer/cron.sh:51]:
```suggestion
    if sudo -u www-data docker ps --format "{{.Names}}" --filter "status=exited" | grep -q "^nextcloud-aio-domaincheck$"; then
```

Comment (Zoey2936) [reply Containers/mastercontainer/cron.sh]: it works with -w and without -w also in my test

**Review by szaimen — APPROVED** [review]
LGTM, thanks!

**Comment by szaimen** [comment]
This is now released with v4.5.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Comment by Zoey2936** [comment]
maybe it would be better to instead enable the autoremove option from docker for this container, while creating it

**Comment by szaimen** [comment]
> maybe it would be better to instead enable the autoremove option from docker for this container, while creating it

Hm... not sure about this. This would make debuggin much harder if it for some reason cannot start correctly, exited and get removed directly without any log or so what went wrong...

**Comment by Zoey2936** [comment]
thats true... but it would be a native docker implementation

**Comment by Zoey2936** [comment]
but will the container not restart if it crashes? since it uses restart: unless-stopped?

**Comment by szaimen** [comment]
> but will the container not restart if it crashes? since it uses restart: unless-stopped?

No it will not but you can still access its logs because it is not removed directly ;)

**Comment by Zoey2936** [comment]
I thought unless-stoped means that container will first be stopped/stopped from restarting until it is manually? or does the container not use unless-stopped?

**Comment by szaimen** [comment]
> or does the container not use unless-stopped?

it does not use unless-stopped

**Comment by Zoey2936** [comment]
Ah, then auto remove makes no sense

**Pull request merged**